src/Discord/Internal/Gateway/Cache.hs

```diff
@@ -20,25 +20,19 @@ data Cache = Cache
      , cacheDMChannels :: !(M.Map ChannelId Channel)
      , cacheGuilds :: !(M.Map GuildId (Maybe (Guild, Maybe GuildCreateData)))
      , cacheChannels :: !(M.Map ChannelId Channel)
-     , cacheApplication :: !PartialApplication
+     , cacheApplication :: !FullApplication
```

Comment:
The [Ready event](https://discord.com/developers/docs/topics/gateway-events#ready) contains a PartialApplication (just id and flags are set) wheras the [GetCurrentApplication](https://discord.com/developers/docs/resources/application) contains all the fields. 

Not sure what to name it exactly. I also considered `Application` (too common? overlap with user types?).